pkg/apis/core/validation/shoot_test.go

```diff
@@ -822,13 +876,27 @@ var _ = Describe("Shoot Validation Tests", func() {
 					"Type":  Equal(field.ErrorTypeInvalid),
 					"Field": Equal("spec.secretBindingName"),
 				})),
+				PointTo(MatchFields(IgnoreExtras, Fields{
+					"Type":  Equal(field.ErrorTypeInvalid),
+					"Field": Equal("spec.credentialsBindingName"),
+				})),
 				PointTo(MatchFields(IgnoreExtras, Fields{
 					"Type":  Equal(field.ErrorTypeInvalid),
 					"Field": Equal("spec.provider.type"),
 				})),
 			))
 		})
 
+		It("should allow setting credentialsBindingName if secretBindingName is nil", func() {
```

Comment:
Nit, simplifies reading:
```suggestion
		It("should allow switching from secretBindingName to credentialsBindingName", func() {
```

Can you add another test case, that validates that the other way around is forbidden?